lib/mintacoin/payments/workers/create_payment.ex

```
@@ -0,0 +1,165 @@
+  @moduledoc """
+  Worker module to perform jobs to create a payment in a blockchain
+  """
+
+  use Oban.Worker, queue: :create_payment_queue, max_attempts: 3
+
+  alias Ecto.{Changeset, UUID}
+
+  alias Mintacoin.{
+    Accounts.Cipher,
+    Assets,
+    Balance,
+    Balances,
+    BlockchainTx,
+    BlockchainTxs,
+    Payments,
+    Payments.Crypto,
+    Payments.Crypto.PaymentResponse,
+    Wallet,
+    Wallets
+  }
+
+  @type amount :: String.t()
+  @type balance :: Balance.t()
+  @type blockchain_tx :: BlockchainTx.t() | Changeset.t() | nil
+  @type id :: UUID.t()
+  @type status :: :ok | :error
+  @type tx_response :: {:ok, PaymentResponse.t()} | {:error, map()}
+
+  @impl true
+  def perform(%Oban.Job{
+        args: %{
+          "source_signature" => source_signature,
+          "source_wallet_id" => source_wallet_id,
+          "destination_wallet_id" => destination_wallet_id,
+          "blockchain_id" => blockchain_id,
+          "asset_id" => asset_id,
+          "amount" => amount,
+          "payment_id" => payment_id
+        }
+      }) do
+    with {:ok, %{code: asset_code}} <- Assets.retrieve_by_id(asset_id),
+         {:ok, %Wallet{public_key: destination_public_key}} <-
+           Wallets.retrieve_by_id(destination_wallet_id),
+         {:ok, %Wallet{encrypted_secret_key: source_encrypted_secret_key}} <-
+           Wallets.retrieve_by_id(source_wallet_id),
+         {:ok, source_secret_key} <- Cipher.decrypt(source_encrypted_secret_key, source_signature) do
+      [
+        source_secret_key: source_secret_key,
+        destination_public_key: destination_public_key,
+        amount: amount,
+        asset_code: asset_code
+      ]
+      |> Crypto.create_payment()
+      |> update_payment_status(payment_id)
+      |> create_blockchain_tx(
+        asset_id,
+        amount,
+        blockchain_id,
+        source_wallet_id,
+        destination_wallet_id,
+        payment_id
+      )
```

move this to a new function and use it `with` for validate the returns

---

@einerzg I think that we shouldn't use a with here, because that is defensive programming, and in order to fix any error that appears in the worker we should let it crash, in this way we can review later and know exactly where and why the error is happening. For that reason, I think is necessary to remove the `with` surrounding this block of code.